src/collection_iterator.cpp

```diff
@@ -27,15 +27,14 @@ bool CollectionIterator::next() {
 }
 
 bool CollectionIterator::decode_value() {
-  DataType::ConstPtr data_type;
   if (collection_->value_type() == CASS_VALUE_TYPE_MAP) {
-    data_type =
+    const DataType::ConstPtr& data_type =
         (index_ % 2 == 0) ? collection_->primary_data_type() : collection_->secondary_data_type();
```

Comment:
BTW if you hid this modulo in your test scenario, change it to
`(index_ & 1) == 0`

Reply:
for performance reasons? it seems all major compilers generate the same code (in isolation at least), so I would prefer to leave that untouched

for example gcc:
https://gcc.godbolt.org/z/W36s4zT53

Reply:
makes sense, it seems that compiler is smart enough, and it is more readable in the original impl